File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse96.java

import javax.crypto.SecretKeyFactory;
import javax.crypto.spec.PBEKeySpec;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.security.spec.InvalidKeySpecException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.util.Base64;
import java.util.HashMap;
import java.util.Map;

public class Lse96 {
    private static final int SALT_LENGTH = 16;
    private static final int ITERATIONS = 10000;
    private static final int KEY_LENGTH = 256;
    private static final String ALGORITHM = "PBKDF2WithHmacSHA256";
    
    // Simulated database
    private static Map<String, String> database = new HashMap<>();
    
    public static String generateSalt() {
        SecureRandom random = new SecureRandom();
        byte[] salt = new byte[SALT_LENGTH];
        random.nextBytes(salt);
        return Base64.getEncoder().encodeToString(salt);
    }
    
    public static String hashPassword(String password, String salt) 
            throws NoSuchAlgorithmException, InvalidKeySpecException {
        byte[] saltBytes = Base64.getDecoder().decode(salt);
        PBEKeySpec spec = new PBEKeySpec(password.toCharArray(), saltBytes, ITERATIONS, KEY_LENGTH);
        SecretKeyFactory factory = SecretKeyFactory.getInstance(ALGORITHM);
        byte[] hash = factory.generateSecret(spec).getEncoded();
        spec.clearPassword();
        return Base64.getEncoder().encodeToString(hash);
    }
    
    public static boolean registerUser(String username, String password) {
        if (username == null || username.trim().isEmpty() || 
            password == null || password.length() < 8) {
            return false;
        }
        
        try {
            String salt = generateSalt();
            String hashedPassword = hashPassword(password, salt);
            String storedValue = salt + ":" + hashedPassword;
            
            // Simulated database storage (in production, use prepared statements)
            database.put(username, storedValue);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }
    
    public static boolean verifyUser(String username, String password) {
        String storedValue = database.get(username);
        if (storedValue == null) {
            return false;
        }
        
        try {
            String[] parts = storedValue.split(":");
            String salt = parts[0];
            String storedHash = parts[1];
            String computedHash = hashPassword(password, salt);
            return storedHash.equals(computedHash);
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }
    
    public static void main(String[] args) {
        // Test Case 1: Valid registration
        System.out.println("Test 1 - Valid registration: " + 
            registerUser("user1", "SecurePass123!"));
        
        // Test Case 2: Valid registration and verification
        registerUser("user2", "MyP@ssw0rd");
        System.out.println("Test 2 - Valid login: " + 
            verifyUser("user2", "MyP@ssw0rd"));
        
        // Test Case 3: Invalid password (too short)
        System.out.println("Test 3 - Short password: " + 
            registerUser("user3", "short"));
        
        // Test Case 4: Invalid login
        registerUser("user4", "ValidPass123");
        System.out.println("Test 4 - Wrong password: " + 
            verifyUser("user4", "WrongPass123"));
        
        // Test Case 5: Empty username
        System.out.println("Test 5 - Empty username: " + 
            registerUser("", "ValidPass123"));
    }
}

File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse96.py

import hashlib
import secrets
import base64
import os
from typing import Optional, Dict

class Lse96:
    SALT_LENGTH = 16
    ITERATIONS = 100000
    HASH_ALGORITHM = 'sha256'
    
    # Simulated database
    database: Dict[str, str] = {}
    
    @staticmethod
    def generate_salt() -> str:
        """Generate a cryptographically secure random salt"""
        salt = secrets.token_bytes(Lse96.SALT_LENGTH)
        return base64.b64encode(salt).decode('utf-8')
    
    @staticmethod
    def hash_password(password: str, salt: str) -> str:
        """Hash password using PBKDF2 with SHA-256"""
        salt_bytes = base64.b64decode(salt)
        password_hash = hashlib.pbkdf2_hmac(
            Lse96.HASH_ALGORITHM,
            password.encode('utf-8'),
            salt_bytes,
            Lse96.ITERATIONS
        )
        return base64.b64encode(password_hash).decode('utf-8')
    
    @staticmethod
    def register_user(username: str, password: str) -> bool:
        """Register a new user with username and password"""
        if not username or not username.strip():
            return False
        if not password or len(password) < 8:
            return False
        
        try:
            salt = Lse96.generate_salt()
            hashed_password = Lse96.hash_password(password, salt)
            stored_value = f"{salt}:{hashed_password}"
            
            # Simulated database storage
            Lse96.database[username] = stored_value
            return True
        except Exception as e:
            print(f"Error: {e}")
            return False
    
    @staticmethod
    def verify_user(username: str, password: str) -> bool:
        """Verify user credentials"""
        stored_value = Lse96.database.get(username)
        if not stored_value:
            return False
        
        try:
            salt, stored_hash = stored_value.split(':')
            computed_hash = Lse96.hash_password(password, salt)
            return secrets.compare_digest(stored_hash, computed_hash)
        except Exception as e:
            print(f"Error: {e}")
            return False

def main():
    # Test Case 1: Valid registration
    print("Test 1 - Valid registration:", 
          Lse96.register_user("user1", "SecurePass123!"))
    
    # Test Case 2: Valid registration and verification
    Lse96.register_user("user2", "MyP@ssw0rd")
    print("Test 2 - Valid login:", 
          Lse96.verify_user("user2", "MyP@ssw0rd"))
    
    # Test Case 3: Invalid password (too short)
    print("Test 3 - Short password:", 
          Lse96.register_user("user3", "short"))
    
    # Test Case 4: Invalid login
    Lse96.register_user("user4", "ValidPass123")
    print("Test 4 - Wrong password:", 
          Lse96.verify_user("user4", "WrongPass123"))
    
    # Test Case 5: Empty username
    print("Test 5 - Empty username:", 
          Lse96.register_user("", "ValidPass123"))

if __name__ == "__main__":
    main()
